scripts/docker/Dockerfile.kokkosllvmproject

```diff
@@ -0,0 +1,53 @@
+FROM nvidia/cuda:10.1-devel
+
+RUN apt-get update && apt-get install -y \
+        bc \
+        git \
+        wget \
+        ccache \
+        python3 \
+        python3-distutils \
+        && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+ARG CMAKE_VERSION=3.10.3
+ENV CMAKE_DIR=/opt/cmake
+RUN CMAKE_KEY=2D2CEF1034921684 && \
+    CMAKE_URL=https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION} && \
+    CMAKE_SCRIPT=cmake-${CMAKE_VERSION}-Linux-x86_64.sh && \
+    CMAKE_SHA256=cmake-${CMAKE_VERSION}-SHA-256.txt && \
+    wget --quiet ${CMAKE_URL}/${CMAKE_SHA256} && \
+    wget --quiet ${CMAKE_URL}/${CMAKE_SHA256}.asc && \
+    wget --quiet ${CMAKE_URL}/${CMAKE_SCRIPT} && \
+    gpg --keyserver pool.sks-keyservers.net --recv-keys ${CMAKE_KEY} && \
+    gpg --verify ${CMAKE_SHA256}.asc ${CMAKE_SHA256} && \
+    grep ${CMAKE_SCRIPT} ${CMAKE_SHA256} | sha256sum --check && \
+    mkdir -p ${CMAKE_DIR} && \
+    sh ${CMAKE_SCRIPT} --skip-license --prefix=${CMAKE_DIR} && \
+    rm cmake*
+ENV PATH=${CMAKE_DIR}/bin:$PATH
+
+ARG NPROC=8
+
+# Clone Kokkos fork of the LLVM Project and build Clang
+ENV LLVM_DIR=/opt/llvm
+RUN LLVM_VERSION=55b3bcf643685c63fcc529d434bed112fdf03939 && \
+    LLVM_URL=https://github.com/kokkos/llvm-project/archive/${LLVM_VERSION}.tar.gz &&\
```

Comment:
Which version is this based on? We shouldn't forgot to document (after merging).

Reply:
I expect this will be handled on the kokkos/llvm-project side and we will use tagged versions from the Dockerfile.

Reply:
I am fine with that.  There is going to be some occasional churn there, but we should be able to deal with it pretty easily.